### PR TITLE
Add AST Builder for document content

### PR DESCRIPTION
**Pull request by nadeeshaan**

### CODE REVIEW

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java [hunk 1]
```diff
@@ -242,4 +278,207 @@ private static Token getDefaultTokenToLeftOrRight(TokenStream tokenStream, int s
...
+                for (Object obj : vars) {
+                    listVarJson.add(obj.toString());
+                }
+            } else if (prop instanceof NodeKind) {
+                String kindName = CaseFormat.UPPER_UNDERSCORE.to(CaseFormat.UPPER_CAMEL, prop.toString());
+                nodeJson.addProperty(jsonName, kindName);
+            } else if (prop instanceof OperatorKind) {
+                nodeJson.addProperty(jsonName, prop.toString());
+
+
```
Comment (NipunaMarcus) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java]: unnecessary new lines

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java [hunk 2]
```diff
@@ -242,4 +278,207 @@ private static Token getDefaultTokenToLeftOrRight(TokenStream tokenStream, int s
...
+                                continue;
+                            }
+                        }
+                        listPropJson.add(generateJSON((Node) listPropItem, anonStructs));
+                    } else {
+                        logger.debug("Can't serialize " + jsonName + ", has a an array of " + listPropItem);
+                    }
+                }
+
+
```
Comment (NipunaMarcus) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java]: unnecessary new lines

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java [hunk 3]
```diff
@@ -242,4 +278,207 @@ private static Token getDefaultTokenToLeftOrRight(TokenStream tokenStream, int s
...
+                    continue;
+                }
+            }
+
+            if (prop instanceof List && jsonName.equals("types")) {
+                // Currently we don't need any Symbols for the UI. So skipping for now.
+                continue;
+            }
+
+
```
Comment (NipunaMarcus) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java]: unnecessary new lines

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java [hunk 4]
```diff
@@ -242,4 +278,207 @@ private static Token getDefaultTokenToLeftOrRight(TokenStream tokenStream, int s
...
+        if (position != null) {
+            JsonObject positionJson = new JsonObject();
+            positionJson.addProperty("startColumn", position.getStartColumn());
+            positionJson.addProperty("startLine", position.getStartLine());
+            positionJson.addProperty("endColumn", position.getEndColumn());
+            positionJson.addProperty("endLine", position.getEndLine());
+            nodeJson.add("position", positionJson);
+        }
+
+        /* Virtual props */
```
Comment (NipunaMarcus) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java]: do we need this?

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java (hunk 1 above)
Comment (nadeeshaan) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java]: Fixed with 69429412db1acf48667f376ed341a05af541dd3a

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java (hunk 2 above)
Comment (nadeeshaan) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java]: Fixed with 69429412db1acf48667f376ed341a05af541dd3a

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java (hunk 3 above)
Comment (nadeeshaan) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java]: Fixed with 69429412db1acf48667f376ed341a05af541dd3a

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java (hunk 4 above)
Comment (nadeeshaan) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java]: Fixed with 69429412db1acf48667f376ed341a05af541dd3a

**Pull request merged**